uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -411,9 +412,15 @@ private static function validateLeaveDatesDoesNotOverlapContracts($params) {
       'end_date' => $toDate->format('Y-m-d'),
     ]);
 
-    if ($contract['count'] > 1) {
+    $contractForEndDate = civicrm_api3('HRJobContract', 'getcontractswithdetailsinperiod', [
```

Comment:
Talked to @tunbola about this, I was confused because of the behavior of the `getcontractswithdetailsinperiod` API call which I would expect returns contracts that start after the $startDate **and** end before the $endDate, but this seems not to be the case. It seems like it returns contracts that start after the $startDate **or** end before the $endDate. This is why a second call is required to validate that the $endDate is in fact inside a valid contract.

I mentioned that the same validation should be done on the $startDate, because as it is it's possible to have a contact with a gap in contracts who applies for a leave request that ends inside a valid contract, but starts where no contract exists.

|--- June ---|  no contract |--- August ---|

- Applies for leave request from 20 July to 2 August
- First API call returns second contract since end date is before its end date
- Second API call returns second contract since end date is before its end date
- Pass validation
- Big explosion with flames everywhere

It might be outside the scope of this, but he will discuss with @davialexandre. 

For now I think we should make it as clear as possible what's happening, maybe I'm just a bit stupid but I was very confused by this. I'll suggest changing `$contract` to `$contractsContainingToOrFromDates` or something similar and `$contractForEndDate` to `$contractsContainingEndDate` because even though ideally it should only be one, it's still a request that can return multiple contracts.